### PR TITLE
add .items() methods to support dict dump

### PR DESCRIPTION
**Pull request by nikolaevigor**
Added methods to support `dict(config.items())` (#55)

Didn't quite get how to make tests for this case since there is no tests/test_config.py, only tests/test_autoconfig.py.

### CODE REVIEW

**Comment by nikolaevigor** [comment]
@henriquebastos can you have a look?

File: decouple.py [hunk 1]
```diff
@@ -138,6 +147,9 @@ def __contains__(self, key):
     def __getitem__(self, key):
         return self.data[key]
 
+    def items(self):
+        return tuple((k, v) for k, v in self.data.items())
```
Comment (henriquebastos) [inline decouple.py]: Why not just return self.data.items()?

File: decouple.py [hunk 2]
```diff
@@ -114,6 +120,9 @@ def __contains__(self, key):
     def __getitem__(self, key):
         return self.parser.get(self.SECTION, key)
 
+    def items(self):
+        return tuple(self.parser.items(self.SECTION))
```
Comment (henriquebastos) [inline decouple.py]: I think we could omit the tuple call.

**Comment by henriquebastos** [comment]
For the tests I think you could simply add a method to test_env.py and test_in.py

File: decouple.py (hunk 1 above)
Comment (nikolaevigor) [reply decouple.py]: self.data.items() returns list of items, when tuple(...) returns tuple of items. Thought can be replaced with tuple(self.data.items()).

File: decouple.py (hunk 2 above)
Comment (nikolaevigor) [reply decouple.py]: Oh, you mean we can return list both places. Ok, will change that.

**Comment by nikolaevigor** [comment]
I've added tests but there was some issues on the way.

1. `RespositoryIni` didn't preserve case sensitivity for keys and values of config. So I've added `self.parser.optionxform = str` in initialization.
2. In Python3 `{}.items()` returns `class <dict_items>` and in Python2 same returns `list`. I've omitted type checking for `RepositoryEnv` in tests. Another options is to cast explicitly `list(self.parser.items())` for `RepositoryEnv.items` method, if we want `RepositoryEnv.items` and `RepositoryIni.items` return same type.

@henriquebastos

**Comment by nikolaevigor** [comment]
@henriquebastos can you have a look?

**Comment by henriquebastos** [comment]
Closing the PR until a concrete use case makes sense.

**Pull request closed without merging**